crates/trie/db/src/state.rs

```diff
@@ -112,26 +112,13 @@ pub trait DatabaseStateRoot<'a, TX>: Sized {
         post_state: &HashedPostStateSorted,
     ) -> Result<(B256, TrieUpdates), StateRootError>;
 
-    /// Calculates the state root for provided [`HashedPostState`] using cached intermediate nodes.
-    fn overlay_root_from_nodes(tx: &'a TX, input: TrieInput) -> Result<B256, StateRootError>;
-
-    /// Calculates the state root and trie updates for provided [`HashedPostState`] using
-    /// cached intermediate nodes.
-    fn overlay_root_from_nodes_with_updates(
-        tx: &'a TX,
-        input: TrieInput,
-    ) -> Result<(B256, TrieUpdates), StateRootError>;
-
     /// Calculates the state root for provided [`HashedPostStateSorted`] using cached intermediate
     /// nodes.
-    fn overlay_root_from_nodes_sorted(
-        tx: &'a TX,
-        input: TrieInputSorted,
-    ) -> Result<B256, StateRootError>;
+    fn overlay_root_from_nodes(tx: &'a TX, input: TrieInputSorted) -> Result<B256, StateRootError>;
```

Comment:
we are changing to make use of `TrieInputSorted` so that 

we can do this
https://github.com/paradigmxyz/reth/pull/20050